<a href="https://colab.research.google.com/github/Aparna10010/NLP-Transformers_Projects/blob/main/TweetSentimentAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Install Required Libraries**

In [ ]:
!pip install transformers
!pip install streamlit
!npm install -g localtunnel

⠙⠹⠸⠼⠴⠦
changed 22 packages in 868ms
⠦
⠦3 packages are looking for funding
⠦  run `npm fund` for details
⠦

# **Import Required Libraries**

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from transformers import pipeline

# **Loading the Pre-trained Sentiment Model & Tokenizer**

In [ ]:
# Load the Sentiment - Analysis pipeline
sentiment_pipeline = pipeline("sentiment-analysis")

# Test the Model
test_text = "I love Delhi but its very polluted not worthy for living"

result = sentiment_pipeline(test_text)
print(result)

No model was supplied, defaulted to distilbert/distilbert-base-uncased-finetuned-sst-2-english and revision 714eb0f (https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english).
Using a pipeline without specifying a model name and revision in production is not recommended.
Device set to use cpu


[{'label': 'NEGATIVE', 'score': 0.9995300769805908}]


- [{'label': 'NEGATIVE', 'score': 0.9995300769805908}]
- This means the model thinks the sentiment is negative with high confidence
- By Default this model will be used :distilbert/distilbert-base-uncased-finetuned-sst-2-english

# **Sentiment Prediction Function**

In [ ]:
# This function will take tweet and returns its sentiment and confidence score:
def get_sentiment(text):
  result = sentiment_pipeline(text)[0]
  label = result['label']
  score = round(result['score'] * 100,2) # convert to percentage
  return label , score


In [ ]:
text = "This Movie was terrible still the cast was amazing but the thing is movie will not be a hit"
label , score = get_sentiment(text)
print(f'Sentiment : {label} , Confidence : {score}%')

Sentiment : NEGATIVE , Confidence : 99.92%


# **Creating Streamlit File (app.py)**

In [ ]:
%%writefile app.py

import streamlit as st
from transformers import pipeline

# Load the Sentiment - Analysis pipeline
sentiment_pipeline = pipeline("sentiment-analysis")

def get_sentiment(text):
  result = sentiment_pipeline(text)[0]
  label = result['label']
  score = round(result['score'] * 100,2) # convert to percentage
  return label , score

# Streamlit UI:

st.set_page_config(page_title = "Tweet Sentiment Analyzer", page_icon = "💬")
st.title("💬 Tweet Sentiment Anlaysis App")
st.write("Enter a text or Tweet below to detect its Sentiment")


user_input = st.text_area("Tweet",
                          placeholder = "Type your Tweet here....")
if st.button("Analyze Sentiment"):
  if user_input.strip() == "":
    st.warning("Please enter some text!")
  else:
    label , score = get_sentiment(user_input)
    st.success(f'**Sentiment:** {label}')
    st.info(f'**Confidence:** {score}%')

Overwriting app.py
